floyd/client/files.py

```diff
@@ -6,70 +6,79 @@
+            # Reset dirs to avoid going further down this directory.
+            # Then continue to the next iteration of os.walk, which causes
+            # everything in this directory to be ignored.
             dirs[:] = []
             floyd_logger.debug("Ignoring directory : {}".format(root))
             continue
 
         for file_name in files:
-            ignore_file = False
-            normalized_path = normalize_path(path, os.path.join(root, file_name))
-            for item in ignore_list_expanded:
-                if PurePath(normalized_path).match(item):
-                    ignore_file = True
-                    break
-
-            if ignore_file:
-                floyd_logger.debug("Ignoring file : {}".format(normalized_path))
+            file_path = unix_stle_path(os.path.join(root, file_name))
+            if ignore_path(file_path, ignore_list, whitelist):
+                floyd_logger.debug("Ignoring file : {}".format(file_name))
                 continue
 
-            file_relative_path = os.path.join(root, file_name)
-            if separator != '/':  # convert relative paths to Unix style
-                file_relative_path = file_relative_path.replace(os.path.sep, '/')
-            file_full_path = os.path.join(os.getcwd(), root, file_name)
+            unignored_files.append(os.path.join(root, file_name))
 
-            local_files.append((file_type, (file_relative_path, open(file_full_path, 'rb'), 'text/plain')))
-            total_file_size += os.path.getsize(file_full_path)
+    return unignored_files
 
-    return (local_files, sizeof_fmt(total_file_size))
+def ignore_path(path, ignore_list=[], whitelist=[]):
+    """
+    Returns a boolean indicating if a path should be ignored given an
+    ignore_list and a whitelist of glob patterns.
+    """
+    return matches_glob_list(path, ignore_list) and not matches_glob_list(path, whitelist)
 
+def matches_glob_list(path, glob_list):
+    """
+    Given a list of glob patterns, returns a boolean
+    indicating if a path matches any glob in the list
+    """
+    for glob in glob_list:
+        try:
+            if PurePath(path).match(glob):
+                return True
+        except TypeError:
+            pass
+    return False
 
-def normalize_path(project_root, path):
+def get_files_in_directory(path, file_type):
     """
-    Convert `path` to a UNIX style path, where `project_root` becomes the root
-    of an imaginery file system (i.e. becomes just an initial "/").
+    Gets the list of files in the directory and subdirectories
+    Respects .floydignore file if present
     """
-    if os.path.sep != '/':
-        path = path.replace(os.path.sep, '/')
-        project_root = project_root.replace(os.path.sep, '/')
+    local_files = []
+    total_file_size = 0
+
+    ignore_list, whitelist = FloydIgnoreManager.get_lists()
 
-    path = path[len(project_root):] if path.startswith(project_root) else path
```

Comment:
Yeah I'm convinced we don't need this anymore. The only difference this will make, given the current design of the cli, is that the string representations of the path of each file that are uploaded to the server will be preceded by a `./`, which doesn't seem to make a difference. In fact, this is the pattern we were following originally, before the most recent changes to this part of the cli.

Things could change if we were to start trying to pass a path other than `"."` to `get_files_in_directory()` or `get_unignored_file_paths()`. The cases in which we would do that seem to be pretty far out of the current workflow, and would open up a number of other edge cases. For example, do we look for a `.floydignore` in parent directories? Do we honor multiple `.floydignore` files in a directory tree? If so, which takes precedence? If we pass a relative path to these methods, do we send the relative path to the API, or do we send them as if they are rooted at `/input`?

I figure that we can address those questions when they come, and I'm reluctant to try to design for them now, since they seem out of the scope of our current direction. I don't know how the old code would have handled these cases, either, but I doubt all the edge cases could have been covered.

If we do decide to just leave it, I think I'll remove the `path` argument from each of those methods, and document them to say that they get the files out of the current directory, since that is really what they are designed to do.

What do you think @narenst?